### DISCUSSION TITLE
`SoftDiceclDiceLoss` and `SoftclDiceLoss` are unclear in their usage

### DISCUSSION THREAD

**Discussion by MathijsdeBoer** [Project-MONAI/MONAI#7590] [Q&A]
I'm doing some 2-class intracranial vessel segmentation experimentation with MonaiLabel. Because the default `DiceCELoss` loss function is not providing satisfactory results, I've been trying to get `SoftDiceclDiceLoss` to work. Unfortunately, the loss values are all over the place, jumping from 4 to -7, hovering around 1.0 for a little while and then flying around again.

I've gone through the available documentation, which is a little sparse, and have also tried gaining some insight from the unit test, which only tests the `torch.ones` case.

Has anyone successfully used either of these loss functions in their projects?

**Comment by KumoLiu** [comment]
Hi @MathijsdeBoer,

Unstable loss values, especially when using `SoftDiceclDiceLoss `, can often be caused by several factors:
Noisy data - Small vessel structures as seen in intracranial vessel segmentation tasks often pose a high complexity due to their thin and intertwined nature. This might cause noisy gradients leading to instability.
Sparse Targets - This loss function might not be suitable for extremely sparse targets (i.e., one class significantly outweighs the other class in the segmentation label), which is often the case in certain medical imaging tasks including vessel segmentation. This can cause the loss value to be rather unstable.
Learning Rate - A high learning rate can cause the loss to fluctuate substantially during training. Try reducing the learning rate to see if it helps in stabilizing the loss.
Yeah, you're right. The documentation for SoftDiceLoss is quite limited.

You could also try using MONAI's GeneralizedDiceLoss which is a proposed modification of the Dice Loss to deal with the issue of class imbalance. In GeneralizedDiceLoss, weights are introduced considering the inverse of each class's volume which results in a more balanced loss.
In MONAI, we can use it like this:
```
from monai.losses import GeneralizedDiceLoss

loss_function = GeneralizedDiceLoss(
    to_onehot_y=True,
    softmax=True,
)
``
It's also often beneficial to experiment and try out different loss functions and see which one works best with your specific data and task.
